zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterServer.java

```diff
@@ -140,10 +136,9 @@ public static void main(String[] args)
 
 
   @Override
-  public void createInterpreter(String interpreterGroupId, String className, Map<String, String>
-          properties)
-      throws TException {
-
+  public void createInterpreter(String interpreterGroupId, String noteId, String
```

Comment:
InterpreterContext always has actual noteId, regardless of "Separate interpreter for each note" is true or false. However, the `noteId` passed to `createInterpreter` is little bit different. 

When "Separate interpreter for each note" true, it is actual noteId. 
However, when "Separate interpreter for each note" is false, the value becomes NoteInterpreterLoader.SHARED_SESSION. So all note shares the same interpreter.